File: python/leetcode11.py
class Solution:
    def maxArea(self, height):
        n = len(height)
        i, j = 0, n-1
        m = 0
        while i < j:
            m = max(m, min(height[j], height[i]) * (j - i))

            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        return m

File: java/leetcode11.java
class Solution {
    public int maxArea(int[] height) {
        int m = 0;
        int i = 0;
        int j = height.length - 1;
        while(i < j){
            m = Math.max(m, Math.min(height[i], height[j]) * (j - i));
            if(height[i] < height[j]) i++;
            else j--;
        }
        return m;
    }
}